qttypes/build.rs

```diff
@@ -64,12 +64,12 @@ fn detect_qreal_size(qt_include_path: &str, qt_library_path: &str) {
 }
 
 fn detect_version_from_header(qt_include_path: &str, qt_library_path: &str) -> String {
-    let f = open_header("qtcoreversion.h", qt_include_path, qt_library_path);
-    let b = BufReader::new(f);
+    const VERSION_HEADER: &'static str = "qtcoreversion.h";
+    let b = open_core_header(VERSION_HEADER, qt_include_path, qt_library_path);
 
     // Find declaration of QTCORE_VERSION_STR
     for line in b.lines() {
-        let line = line.expect("qtcoreversion.h is valid UTF-8");
+        let line = line.expect(&format!("UTF-8 conversion failed for `{:?}`", VERSION_HEADER));
```

Comment:
nitpick:  format in expect is ineficcient because the allocation and string formating will happen no matter what. 
maybe the best instead is to use a `.or_else(|| panic!(...))`  construct

Reply:
Yeah, I didn't notice it's in the loop. I thought, like, who cares about build script micro-optimizations. But this is exactly not the case.

Reply:
Reverted to a plain `expect` with a message like in this `format`.